pywbem/_logging.py

```
@@ -0,0 +1,402 @@
+                    log_level = 'debug'
+            elif log_component == 'ops':
+                logger_name = LOG_OPS_CALLS_NAME
+                if log_level is None:
+                    log_level = 'debug'
+            else:
+                raise ValueError('Invalid log_component %s' % log_component)
+
+            # Set the log level based on the log_level input
+            level = getattr(logging, log_level.upper(), None)
+            # check logging log_level_choices have not changed from expected
+            assert isinstance(level, int)
+            if level is None:
+                raise ValueError('Invalid log level %s specified. Must be one '
+                                 'of %s.' % (log_level, LOG_LEVELS))
+
+            # create named logger
+            if handler:
+                handler.setFormatter(logging.Formatter(format_string))
+                logger = logging.getLogger(logger_name)
+                logger.addHandler(handler)
+                logger.setLevel(level)
+
+            # save the detail level in the dict that is part of this class.
+            # All members of this tuple are just viewing except detail
+            # that is used by individual loggers
+            cls.loggers[logger_name] = (log_detail_level, log_level,
+                                        log_dest, log_filename)
+
+    @classmethod
+    def get_logger(cls, logger_name):
+        """
+            Get the name of a logger that has been defined.
+        """
+        if logger_name in cls.loggers:
+            return cls.loggers[logger_name][0]
+        return None
+
+    @classmethod
+    def get_logger_info(cls, logger_name):
+        """Get information about a logger by name"""
+        if logger_name in cls.loggers:
+            t = cls.loggers[logger_name]
+            return 'comp=%s dest=%s detail=%s log_level=%s ' \
+                   'file=%s' % (logger_name, t[3], t[0], t[2], t[3])
+        return('%s not defined.' % logger_name)
```

I suggest that the return value of this method is changed to be a tuple, and not a string. If the logger is not found, let it raise KeyError (as a byproduct of accessing the dict without check).